pages/doc/integrations_tas_custom_app_metrics_howto.md

```diff
@@ -0,0 +1,75 @@
+---
+title: How to Monitor Custom App Metrics on TAS with Tanzu Observability
+keywords:
+tags: [integrations]
+sidebar: doc_sidebar
+permalink: integrations_tas_custom_app_metrics_howto.html
+summary: How to get your custom application metrics to appear in Tanzu Observability by Wavefront.
+---
+
+Out of the box, the Wavefront Nozzle for TAS provides metrics for monitoring the TAS platform. For monitoring applications running **on** the platform, the Nozzle supports two approaches:
+
+1. **Metrics Registrar:** a **pull-based** model based on the **Prometheus metrics format**
+1. **Wavefront Service Broker:** a **push-based** model where applications send metrics directly to a Wavefront Proxy.
+
+**We recommend the Metrics Registrar approach** as a simpler and more predictable pattern, but teams may choose either option. Both have drawbacks, and for some workloads only the push method can be used in practice.
+
+## Push vs Pull Models Generally
+
+The “pull” or “scrape” model has gained popularity in recent years. The typical pull-based design involves hosting an HTTP route on your server, typically at `/metrics`, which exposes custom metrics. Another process periodically fetches the response from that route and transforms it into metrics within a metric store.
```

Comment:
The “pull” or “scrape” model has gained popularity in recent years. The common pull-based design involves hosting an HTTP route on your server, typically at `/metrics`, which exposes custom metrics. Another process periodically fetches the response from that route and transforms it into metrics within a metric store.

In the push model, the application periodically sends metrics directly to a server by using authentication credentials provided to the application.

Scraping is a popular approach, because the application being monitored does not need to know anything about the metrics store, such as an endpoint or a credential.

In the scraping model, the rate of metrics collection can be centrally controlled. In a push-based model, applications may or may not be able to throttle the metric volume if they are overwhelming the metrics service.

The output of a `/metrics` endpoint in a scraping model is normally human readable, which can help with debugging the metrics coming from a running application.

Pushing metrics can produce more details. In the push-based model, a metric can be submitted as often as practically limits allow - possibly multiple times per second. In the scrape model, only changes since the last scrape are seen. Typically, to be useful metrics data does not need to be extremely high-fidelity, and this periodic sampling is acceptable to most users.

Short-lived processes **must** use a push-based model, because they do not run long enough to be discovered and scraped by a metrics collector.